### ISSUE TITLE
Loss Change

### ISSUE THREAD

**Issue by BingCS** [wohlert/generative-query-network-pytorch#19]
Dear wohlert,

May I consult you several questions?

1. I tried to train this network on Mazes Data from https://github.com/deepmind/gqn-datasets. Actually it just contains 5% data, which is around 110000, instead of the full data. Is it right?

2. I trained 30000 steps, but the elbo loss only converged to 6800 which has a big difference compared to around 7 in the supplementary. So may I ask what is the approximate value do you achieve on the data you used?

3. From the visualisation based on Question 2, the reconstruction seems to be reasonable. But the sampling results is quite bad. Do you meet the same problem?

Many thanks,
Bing

**Comment by wohlert** [comment]
Hello.

1. All of the data should be in the gqn-datasets storage, but I do not know how you're implementing the dataloader. If you are modifying my code, then note that one of the options is to only take a fraction of the data. This might be what you have done.

2. This gap could also explain the poor generations. If you are annealing according to the paper, then keep in mind that this works only if you have the whole dataset. You might need to change the learning rate and sigma scheme to fit the amount.

3. Unconditional generation (without any context) is always quite bad, even in the DeepMind paper. With more information comes better generations.

I suspect all of your problems are due to not using enough data.

**Comment by BingCS** [comment]
Dear Wohlert,

Many thanks for your kind reply.

I check the gqn dataset from https://github.com/deepmind/gqn-datasets. Shepard_metzler_7_parts contains 900 tfrecords for training. Each tfrecord has 20 scenes. So there are only 18000 scenes. For Mazes, there are 1080 records with 100 scenes for each record, which means 108000 in total.

So I think this link (https://github.com/deepmind/gqn-datasets) just contains a part of the whole data. Did you get the whole data from that link? 

Best,
Bing

**Comment by wohlert** [comment]
If there are 108000 scenes, then it should be the whole dataset, so I do not know what you are referring to. All of the data should be available in the deepmind repository.

**Comment by BingCS** [comment]
Thanks. But for Mazes, they generated 2 million scenes with 300 images for each scene according to the paper. So why it's the whole dataset?

**Comment by wohlert** [comment]
I think you will have to ask the authors instead. That question is not relevant to this repository.
